hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -612,6 +614,7 @@ public void readData(ObjectDataInput in) throws IOException {
         // RU_COMPAT_4_2
         if (in.getVersion().isGreaterOrEqual(Versions.V5_0)) {
             merkleTreeConfig = in.readObject();
+            dataPersistenceConfig = in.readObject();
```

Comment:
isn't setting only for one enough? So `setDataPersistenceConfig(in.readObject())` does trigger merge and if `dataPersistence` is not read it would mean merge operation is not necessary

Reply:
Then we need `setHotRestartConfig()` in the `else` block. Or we need to call `merge...()` unconditionally at the end of the deserialization. The point is we can deserialize an object with only HR config filled (if the member serialized the object is on version `<5.0`) and after deserialization, we want to have the merging done.

Reply:
ohh, I see, thank you :)